### ISSUE TITLE
Validation of KBV_PR_EAU_Bundle|1.0.2 fails for official example Bundles

### ISSUE THREAD

**Issue by fleischa** [gematik/app-referencevalidator#3]
The validation of the official KBV example Bundles found on simplifier.net [(https://simplifier.net/packages/kbv.ita.eau/1.0.2/files/319379)](https://simplifier.net/packages/kbv.ita.eau/1.0.2/files/319379) fails with validator version 0.3.0.
```
16.03.23-11:10:49.704 [main] INFO  ca.uhn.fhir.util.VersionUtil  - HAPI FHIR version 6.2.4 - Rev 285043d7f4
16.03.23-11:10:49.709 [main] INFO  ca.uhn.fhir.context.FhirContext  - Creating new FHIR context for FHIR version [R4]
16.03.23-11:10:49.875 [main] INFO  de.gematik.refv.valmodule.eau.EauValidationModule  - Reading input file 08b97c0c-6b65-4f33-a985-66ffd6fb2322.xml...
16.03.23-11:10:49.901 [main] INFO  de.gematik.refv.commons.validation.GenericValidator  - Validating against https://fhir.kbv.de/StructureDefinition/KBV_PR_EAU_Bundle|1.0.2...
16.03.23-11:10:51.582 [main] INFO  ca.uhn.fhir.validation.FhirValidator  - Ph-schematron library not found on classpath, will not attempt to perform schematron validation
16.03.23-11:10:51.585 [main] INFO  ca.uhn.fhir.context.FhirContext  - Creating new FHIR context for FHIR version [R5]
16.03.23-11:10:51.663 [main] INFO  ca.uhn.fhir.context.support.DefaultProfileValidationSupport  - Loading structure definitions from classpath: /org/hl7/fhir/r4/model/profile/profiles-resources.xml
16.03.23-11:10:51.687 [main] INFO  ca.uhn.fhir.util.XmlUtil  - FHIR XML procesing will use StAX implementation 'null' version 'null'
16.03.23-11:10:52.135 [main] INFO  ca.uhn.fhir.context.support.DefaultProfileValidationSupport  - Loading structure definitions from classpath: /org/hl7/fhir/r4/model/profile/profiles-types.xml
16.03.23-11:10:52.173 [main] INFO  ca.uhn.fhir.context.support.DefaultProfileValidationSupport  - Loading structure definitions from classpath: /org/hl7/fhir/r4/model/profile/profiles-others.xml
16.03.23-11:10:52.272 [main] INFO  ca.uhn.fhir.context.support.DefaultProfileValidationSupport  - Loading structure definitions from classpath: /org/hl7/fhir/r4/model/extension/extension-definitions.xml
16.03.23-11:10:53.424 [main] INFO  ca.uhn.fhir.context.support.DefaultProfileValidationSupport  - Loading CodeSystem/ValueSet from classpath: /org/hl7/fhir/r4/model/valueset/valuesets.xml
16.03.23-11:10:53.569 [main] INFO  ca.uhn.fhir.context.support.DefaultProfileValidationSupport  - Loading CodeSystem/ValueSet from classpath: /org/hl7/fhir/r4/model/valueset/v2-tables.xml
16.03.23-11:10:53.654 [main] INFO  ca.uhn.fhir.context.support.DefaultProfileValidationSupport  - Loading CodeSystem/ValueSet from classpath: /org/hl7/fhir/r4/model/valueset/v3-codesystems.xml
16.03.23-11:10:54.135 [main] INFO  de.gematik.refv.commons.validation.GenericValidator  - Final ValidationResult: Valid: false. Messages: 
SingleValidationMessage[col=28,row=64,locationString=Bundle.entry[0].resource.ofType(Composition).section[0].entry[0],message=Unable to find a match for profile Condition/a246b026-c47c-4257-abed-9887ed7996eb among choices: https://fhir.kbv.de/StructureDefinition/KBV_PR_EAU_Condition_ICD|1.0.2,Reference_REF_CantMatchChoice,severity=error]
SingleValidationMessage[col=28,row=64,locationString=Bundle.entry[0].resource.ofType(Composition).section[0].entry[0],message=Details for Condition/a246b026-c47c-4257-abed-9887ed7996eb matching against Profile https://fhir.kbv.de/StructureDefinition/KBV_PR_EAU_Condition_ICD,severity=information]
SingleValidationMessage[col=28,row=75,locationString=Bundle.entry[0].resource.ofType(Composition).section[1].entry[0],message=Unable to find a match for profile Condition/854ff3b2-f0ca-41f9-9bd1-0204f51a847e among choices: https://fhir.kbv.de/StructureDefinition/KBV_PR_EAU_Condition_ICD|1.0.2,Reference_REF_CantMatchChoice,severity=error]
SingleValidationMessage[col=28,row=75,locationString=Bundle.entry[0].resource.ofType(Composition).section[1].entry[0],message=Details for Condition/854ff3b2-f0ca-41f9-9bd1-0204f51a847e matching against Profile https://fhir.kbv.de/StructureDefinition/KBV_PR_EAU_Condition_ICD,severity=information]
SingleValidationMessage[col=28,row=86,locationString=Bundle.entry[0].resource.ofType(Composition).section[2].entry[0],message=Unable to find a match for profile Condition/065b2388-e4da-40ac-a459-2f392a0a41d9 among choices: https://fhir.kbv.de/StructureDefinition/KBV_PR_EAU_Condition_ICD|1.0.2,Reference_REF_CantMatchChoice,severity=error]
SingleValidationMessage[col=28,row=86,locationString=Bundle.entry[0].resource.ofType(Composition).section[2].entry[0],message=Details for Condition/065b2388-e4da-40ac-a459-2f392a0a41d9 matching against Profile https://fhir.kbv.de/StructureDefinition/KBV_PR_EAU_Condition_ICD,severity=information]
SingleValidationMessage[col=28,row=97,locationString=Bundle.entry[0].resource.ofType(Composition).section[3].entry[0],message=Unable to find a match for profile Condition/bc96a462-2ecf-4c63-9b22-bc02bb8f1155 among choices: https://fhir.kbv.de/StructureDefinition/KBV_PR_EAU_Condition_ICD|1.0.2,Reference_REF_CantMatchChoice,severity=error]
SingleValidationMessage[col=28,row=97,locationString=Bundle.entry[0].resource.ofType(Composition).section[3].entry[0],message=Details for Condition/bc96a462-2ecf-4c63-9b22-bc02bb8f1155 matching against Profile https://fhir.kbv.de/StructureDefinition/KBV_PR_EAU_Condition_ICD,severity=information]
SingleValidationMessage[col=28,row=108,locationString=Bundle.entry[0].resource.ofType(Composition).section[4].entry[0],message=Unable to find a match for profile Condition/0d27fa35-55d9-48d0-b42e-d3bb3f85edfb among choices: https://fhir.kbv.de/StructureDefinition/KBV_PR_EAU_Condition_ICD|1.0.2,Reference_REF_CantMatchChoice,severity=error]
SingleValidationMessage[col=28,row=108,locationString=Bundle.entry[0].resource.ofType(Composition).section[4].entry[0],message=Details for Condition/0d27fa35-55d9-48d0-b42e-d3bb3f85edfb matching against Profile https://fhir.kbv.de/StructureDefinition/KBV_PR_EAU_Condition_ICD,severity=information]
SingleValidationMessage[col=28,row=119,locationString=Bundle.entry[0].resource.ofType(Composition).section[5].entry[0],message=Unable to find a match for profile Condition/35a2833d-0cbd-49d4-86f8-c97a64154890 among choices: https://fhir.kbv.de/StructureDefinition/KBV_PR_EAU_Condition_ICD|1.0.2,Reference_REF_CantMatchChoice,severity=error]
SingleValidationMessage[col=28,row=119,locationString=Bundle.entry[0].resource.ofType(Composition).section[5].entry[0],message=Details for Condition/35a2833d-0cbd-49d4-86f8-c97a64154890 matching against Profile https://fhir.kbv.de/StructureDefinition/KBV_PR_EAU_Condition_ICD,severity=information]
SingleValidationMessage[col=29,row=442,locationString=Bundle.entry[6].resource.ofType(Condition).code.coding[0],message=Code system has been ignored due to module configuration for 'http://fhir.de/CodeSystem/dimdi/icd-10-gm#E10.20',Terminology_PassThrough_TX_Message,severity=information]
SingleValidationMessage[col=106,row=443,locationString=Bundle.entry[6].resource.ofType(Condition).code.coding[0].extension[0],message=Slicing cannot be evaluated: Unable to resolve profile CanonicalType[http://fhir.de/StructureDefinition/seitenlokalisation|0.9.13],SLICING_CANNOT_BE_EVALUATED,severity=error]
SingleValidationMessage[col=29,row=484,locationString=Bundle.entry[7].resource.ofType(Condition).code.coding[0],message=Code system has been ignored due to module configuration for 'http://fhir.de/CodeSystem/dimdi/icd-10-gm#N08.3',Terminology_PassThrough_TX_Message,severity=information]
SingleValidationMessage[col=106,row=485,locationString=Bundle.entry[7].resource.ofType(Condition).code.coding[0].extension[0],message=Slicing cannot be evaluated: Unable to resolve profile CanonicalType[http://fhir.de/StructureDefinition/seitenlokalisation|0.9.13],SLICING_CANNOT_BE_EVALUATED,severity=error]
SingleValidationMessage[col=29,row=526,locationString=Bundle.entry[8].resource.ofType(Condition).code.coding[0],message=Code system has been ignored due to module configuration for 'http://fhir.de/CodeSystem/dimdi/icd-10-gm#S42.3',Terminology_PassThrough_TX_Message,severity=information]
SingleValidationMessage[col=106,row=527,locationString=Bundle.entry[8].resource.ofType(Condition).code.coding[0].extension[0],message=Slicing cannot be evaluated: Unable to resolve profile CanonicalType[http://fhir.de/StructureDefinition/seitenlokalisation|0.9.13],SLICING_CANNOT_BE_EVALUATED,severity=error]
SingleValidationMessage[col=96,row=533,locationString=Bundle.entry[8].resource.ofType(Condition).code.coding[0].extension[1],message=Slicing cannot be evaluated: Unable to resolve profile CanonicalType[http://fhir.de/StructureDefinition/seitenlokalisation|0.9.13],SLICING_CANNOT_BE_EVALUATED,severity=error]
SingleValidationMessage[col=29,row=574,locationString=Bundle.entry[9].resource.ofType(Condition).code.coding[0],message=Code system has been ignored due to module configuration for 'http://fhir.de/CodeSystem/dimdi/icd-10-gm#S82.28',Terminology_PassThrough_TX_Message,severity=information]
SingleValidationMessage[col=106,row=575,locationString=Bundle.entry[9].resource.ofType(Condition).code.coding[0].extension[0],message=Slicing cannot be evaluated: Unable to resolve profile CanonicalType[http://fhir.de/StructureDefinition/seitenlokalisation|0.9.13],SLICING_CANNOT_BE_EVALUATED,severity=error]
SingleValidationMessage[col=96,row=581,locationString=Bundle.entry[9].resource.ofType(Condition).code.coding[0].extension[1],message=Slicing cannot be evaluated: Unable to resolve profile CanonicalType[http://fhir.de/StructureDefinition/seitenlokalisation|0.9.13],SLICING_CANNOT_BE_EVALUATED,severity=error]
SingleValidationMessage[col=29,row=622,locationString=Bundle.entry[10].resource.ofType(Condition).code.coding[0],message=Code system has been ignored due to module configuration for 'http://fhir.de/CodeSystem/dimdi/icd-10-gm#Q01.9',Terminology_PassThrough_TX_Message,severity=information]
SingleValidationMessage[col=106,row=623,locationString=Bundle.entry[10].resource.ofType(Condition).code.coding[0].extension[0],message=Slicing cannot be evaluated: Unable to resolve profile CanonicalType[http://fhir.de/StructureDefinition/seitenlokalisation|0.9.13],SLICING_CANNOT_BE_EVALUATED,severity=error]
SingleValidationMessage[col=29,row=664,locationString=Bundle.entry[11].resource.ofType(Condition).code.coding[0],message=Code system has been ignored due to module configuration for 'http://fhir.de/CodeSystem/dimdi/icd-10-gm#S22.32',Terminology_PassThrough_TX_Message,severity=information]
SingleValidationMessage[col=106,row=665,locationString=Bundle.entry[11].resource.ofType(Condition).code.coding[0].extension[0],message=Slicing cannot be evaluated: Unable to resolve profile CanonicalType[http://fhir.de/StructureDefinition/seitenlokalisation|0.9.13],SLICING_CANNOT_BE_EVALUATED,severity=error]

16.03.23-11:10:54.140 [main] INFO  de.gematik.refv.cli.ReferenceValidator  - Validation result: Valid: false. Messages:
SingleValidationMessage[col=28,row=64,locationString=Bundle.entry[0].resource.ofType(Composition).section[0].entry[0],message=Unable to find a match for profile Condition/a246b026-c47c-4257-abed-9887ed7996eb among choices: https://fhir.kbv.de/StructureDefinition/KBV_PR_EAU_Condition_ICD|1.0.2,Reference_REF_CantMatchChoice,severity=error]
SingleValidationMessage[col=28,row=64,locationString=Bundle.entry[0].resource.ofType(Composition).section[0].entry[0],message=Details for Condition/a246b026-c47c-4257-abed-9887ed7996eb matching against Profile https://fhir.kbv.de/StructureDefinition/KBV_PR_EAU_Condition_ICD,severity=information]
SingleValidationMessage[col=28,row=75,locationString=Bundle.entry[0].resource.ofType(Composition).section[1].entry[0],message=Unable to find a match for profile Condition/854ff3b2-f0ca-41f9-9bd1-0204f51a847e among choices: https://fhir.kbv.de/StructureDefinition/KBV_PR_EAU_Condition_ICD|1.0.2,Reference_REF_CantMatchChoice,severity=error]
SingleValidationMessage[col=28,row=75,locationString=Bundle.entry[0].resource.ofType(Composition).section[1].entry[0],message=Details for Condition/854ff3b2-f0ca-41f9-9bd1-0204f51a847e matching against Profile https://fhir.kbv.de/StructureDefinition/KBV_PR_EAU_Condition_ICD,severity=information]
SingleValidationMessage[col=28,row=86,locationString=Bundle.entry[0].resource.ofType(Composition).section[2].entry[0],message=Unable to find a match for profile Condition/065b2388-e4da-40ac-a459-2f392a0a41d9 among choices: https://fhir.kbv.de/StructureDefinition/KBV_PR_EAU_Condition_ICD|1.0.2,Reference_REF_CantMatchChoice,severity=error]
SingleValidationMessage[col=28,row=86,locationString=Bundle.entry[0].resource.ofType(Composition).section[2].entry[0],message=Details for Condition/065b2388-e4da-40ac-a459-2f392a0a41d9 matching against Profile https://fhir.kbv.de/StructureDefinition/KBV_PR_EAU_Condition_ICD,severity=information]
SingleValidationMessage[col=28,row=97,locationString=Bundle.entry[0].resource.ofType(Composition).section[3].entry[0],message=Unable to find a match for profile Condition/bc96a462-2ecf-4c63-9b22-bc02bb8f1155 among choices: https://fhir.kbv.de/StructureDefinition/KBV_PR_EAU_Condition_ICD|1.0.2,Reference_REF_CantMatchChoice,severity=error]
SingleValidationMessage[col=28,row=97,locationString=Bundle.entry[0].resource.ofType(Composition).section[3].entry[0],message=Details for Condition/bc96a462-2ecf-4c63-9b22-bc02bb8f1155 matching against Profile https://fhir.kbv.de/StructureDefinition/KBV_PR_EAU_Condition_ICD,severity=information]
SingleValidationMessage[col=28,row=108,locationString=Bundle.entry[0].resource.ofType(Composition).section[4].entry[0],message=Unable to find a match for profile Condition/0d27fa35-55d9-48d0-b42e-d3bb3f85edfb among choices: https://fhir.kbv.de/StructureDefinition/KBV_PR_EAU_Condition_ICD|1.0.2,Reference_REF_CantMatchChoice,severity=error]
SingleValidationMessage[col=28,row=108,locationString=Bundle.entry[0].resource.ofType(Composition).section[4].entry[0],message=Details for Condition/0d27fa35-55d9-48d0-b42e-d3bb3f85edfb matching against Profile https://fhir.kbv.de/StructureDefinition/KBV_PR_EAU_Condition_ICD,severity=information]
SingleValidationMessage[col=28,row=119,locationString=Bundle.entry[0].resource.ofType(Composition).section[5].entry[0],message=Unable to find a match for profile Condition/35a2833d-0cbd-49d4-86f8-c97a64154890 among choices: https://fhir.kbv.de/StructureDefinition/KBV_PR_EAU_Condition_ICD|1.0.2,Reference_REF_CantMatchChoice,severity=error]
SingleValidationMessage[col=28,row=119,locationString=Bundle.entry[0].resource.ofType(Composition).section[5].entry[0],message=Details for Condition/35a2833d-0cbd-49d4-86f8-c97a64154890 matching against Profile https://fhir.kbv.de/StructureDefinition/KBV_PR_EAU_Condition_ICD,severity=information]
SingleValidationMessage[col=29,row=442,locationString=Bundle.entry[6].resource.ofType(Condition).code.coding[0],message=Code system has been ignored due to module configuration for 'http://fhir.de/CodeSystem/dimdi/icd-10-gm#E10.20',Terminology_PassThrough_TX_Message,severity=information]
SingleValidationMessage[col=106,row=443,locationString=Bundle.entry[6].resource.ofType(Condition).code.coding[0].extension[0],message=Slicing cannot be evaluated: Unable to resolve profile CanonicalType[http://fhir.de/StructureDefinition/seitenlokalisation|0.9.13],SLICING_CANNOT_BE_EVALUATED,severity=error]
SingleValidationMessage[col=29,row=484,locationString=Bundle.entry[7].resource.ofType(Condition).code.coding[0],message=Code system has been ignored due to module configuration for 'http://fhir.de/CodeSystem/dimdi/icd-10-gm#N08.3',Terminology_PassThrough_TX_Message,severity=information]
SingleValidationMessage[col=106,row=485,locationString=Bundle.entry[7].resource.ofType(Condition).code.coding[0].extension[0],message=Slicing cannot be evaluated: Unable to resolve profile CanonicalType[http://fhir.de/StructureDefinition/seitenlokalisation|0.9.13],SLICING_CANNOT_BE_EVALUATED,severity=error]
SingleValidationMessage[col=29,row=526,locationString=Bundle.entry[8].resource.ofType(Condition).code.coding[0],message=Code system has been ignored due to module configuration for 'http://fhir.de/CodeSystem/dimdi/icd-10-gm#S42.3',Terminology_PassThrough_TX_Message,severity=information]
SingleValidationMessage[col=106,row=527,locationString=Bundle.entry[8].resource.ofType(Condition).code.coding[0].extension[0],message=Slicing cannot be evaluated: Unable to resolve profile CanonicalType[http://fhir.de/StructureDefinition/seitenlokalisation|0.9.13],SLICING_CANNOT_BE_EVALUATED,severity=error]
SingleValidationMessage[col=96,row=533,locationString=Bundle.entry[8].resource.ofType(Condition).code.coding[0].extension[1],message=Slicing cannot be evaluated: Unable to resolve profile CanonicalType[http://fhir.de/StructureDefinition/seitenlokalisation|0.9.13],SLICING_CANNOT_BE_EVALUATED,severity=error]
SingleValidationMessage[col=29,row=574,locationString=Bundle.entry[9].resource.ofType(Condition).code.coding[0],message=Code system has been ignored due to module configuration for 'http://fhir.de/CodeSystem/dimdi/icd-10-gm#S82.28',Terminology_PassThrough_TX_Message,severity=information]
SingleValidationMessage[col=106,row=575,locationString=Bundle.entry[9].resource.ofType(Condition).code.coding[0].extension[0],message=Slicing cannot be evaluated: Unable to resolve profile CanonicalType[http://fhir.de/StructureDefinition/seitenlokalisation|0.9.13],SLICING_CANNOT_BE_EVALUATED,severity=error]
SingleValidationMessage[col=96,row=581,locationString=Bundle.entry[9].resource.ofType(Condition).code.coding[0].extension[1],message=Slicing cannot be evaluated: Unable to resolve profile CanonicalType[http://fhir.de/StructureDefinition/seitenlokalisation|0.9.13],SLICING_CANNOT_BE_EVALUATED,severity=error]
SingleValidationMessage[col=29,row=622,locationString=Bundle.entry[10].resource.ofType(Condition).code.coding[0],message=Code system has been ignored due to module configuration for 'http://fhir.de/CodeSystem/dimdi/icd-10-gm#Q01.9',Terminology_PassThrough_TX_Message,severity=information]
SingleValidationMessage[col=106,row=623,locationString=Bundle.entry[10].resource.ofType(Condition).code.coding[0].extension[0],message=Slicing cannot be evaluated: Unable to resolve profile CanonicalType[http://fhir.de/StructureDefinition/seitenlokalisation|0.9.13],SLICING_CANNOT_BE_EVALUATED,severity=error]
SingleValidationMessage[col=29,row=664,locationString=Bundle.entry[11].resource.ofType(Condition).code.coding[0],message=Code system has been ignored due to module configuration for 'http://fhir.de/CodeSystem/dimdi/icd-10-gm#S22.32',Terminology_PassThrough_TX_Message,severity=information]
SingleValidationMessage[col=106,row=665,locationString=Bundle.entry[11].resource.ofType(Condition).code.coding[0].extension[0],message=Slicing cannot be evaluated: Unable to resolve profile CanonicalType[http://fhir.de/StructureDefinition/seitenlokalisation|0.9.13],SLICING_CANNOT_BE_EVALUATED,severity=error]
```
To me it appears as if the main problem is a mismatched version of `http://fhir.de/StructureDefinition/seitenlokalisation`. The validator attempts to look up version 0.9.13 while the jar contains version 1.0.0.

**Comment by alexey-tschudnowsky** [comment]
Hi @fleischa , 
thank you for the report. Indeed this is a bug in the EAU validation module and it will be fixed with the coming 0.4 release

**Comment by fleischa** [comment]
Hello @alexey-tschudnowsky,
awesome, thanks for the prompt reply.

**Comment by alexey-tschudnowsky** [comment]
The issue is resolved in the [0.4.1-Release](https://github.com/gematik/app-referencevalidator/releases/tag/0.4.1)
